src/konfo_backend/search/query.clj

```diff
@@ -16,23 +16,56 @@
 
 (defn- some-hakuaika-kaynnissa
   []
-  {:nested {:path "hits.hakuajat" :query {:bool {:filter [{:range {:hits.hakuajat.alkaa {:lte (current-time-as-kouta-format)}}}
-                                                          {:bool  {:should [{:bool {:must_not {:exists {:field "hits.hakuajat.paattyy"}}}},
-                                                                            {:range {:hits.hakuajat.paattyy {:gt (current-time-as-kouta-format)}}}]}}]}}}})
+  { :nested {:path "hits.hakutiedot.hakuajat"
+             :query {:bool {:filter [{:range {:hits.hakutiedot.hakuajat.alkaa {:lte (current-time-as-kouta-format)}}}
+                                     {:bool  {:should [{:bool {:must_not {:exists {:field "hits.hakutiedot.hakuajat.paattyy"}}}},
+                                                       {:range {:hits.hakutiedot.hakuajat.paattyy {:gt (current-time-as-kouta-format)}}}]}}]}}}})
+
+(defn- some-past-hakuaika-still-viable
+  []
+  { :nested {:path "hits.hakutiedot.hakuajat"
+             :query {:bool {:should [{:bool {:must_not {:exists {:field "hits.hakutiedot.hakuajat.paattyy"}}}},
+                                     {:range {:hits.hakutiedot.hakuajat.paattyy {:gt (half-year-past-as-kouta-format)}}}]}}}})
+
+; NOTE Rajattavat hakutiedot täytyy yhdistää hakuaika-käynnissä rajaukseen poissulkevasti ( AND ) aikarajan perusteella (käynnissä vs 6kk vanhat)
+(defn- hakutieto-query
+  ([inner-query]
+   {:nested {:path "hits.hakutiedot"
+             :query { :bool { :filter inner-query }}}})
+  ([haku-kaynnissa inner-query]
+   {:nested {:path "hits.hakutiedot"
+             :query { :bool { :filter [(if haku-kaynnissa
+                                         (some-hakuaika-kaynnissa)
+                                         (some-past-hakuaika-still-viable))
+                                       inner-query] }}}}))
+
+(defn- is-haku-kaynnissa-not-already-included
```

Comment:
Tämän nimi olisi clojurelle tyypillisellä tavalla haku-kaynnissa-not-already-included? eli boolean arvon palauttavissa funktioissa olisi kysymysmerkki lopussa mutta eipä itsellä ole vahvoja mielipiteitä tuosta suuntaan tai toiseen. Toisaalta se on käytäntö mitä ihan clojure.core käyttää.